File: src/main/java/com/marsraver/LedFx/wled/WledDevice.java
package com.marsraver.LedFx.wled;

public class WledDevice {
    public String ip;
    public String name;
    public String version;
    public int ledCount;
    public long uptime;
    public boolean isOn;

    @Override
    public String toString() {
        String power = isOn ? "ON" : "OFF";
        return String.format("  • WLED '%s' (%s) – version %s, %d LEDs, uptime %ds, Power: %s",
                name, ip, version, ledCount, uptime, power);
    }
}

File: src/main/kotlin/com/marsraver/LedFx/wled/WledDevice.kt
package com.marsraver.LedFx.wled

class WledDevice {
    var ip: String? = null
    var name: String? = null
    var version: String? = null
    var ledCount: Int = 0
    var uptime: Long = 0
    var isOn: Boolean = false

    override fun toString(): String {
        val power = if (isOn) "ON" else "OFF"
        return String.format(
            "  • WLED '%s' (%s) – version %s, %d LEDs, uptime %ds, Power: %s",
            name, ip, version, ledCount, uptime, power
        )
    }
}

